src/ert/ensemble_evaluator/snapshot.py

```diff
@@ -104,110 +102,175 @@ def convert_iso8601_to_datetime(
     return parse(timestamp)
 
 
+DICT_SEP = "/"
+
+
+def _filter_nones(some_dict: dict) -> dict:
+    return {key: value for key, value in some_dict.items() if value is not None}
+
+
 class PartialSnapshot:
-    def __init__(self, snapshot: "Snapshot") -> None:
-        """Create a PartialSnapshot. If no snapshot is provided, the object is
-        a immutable POD, and any attempt at mutating it will raise an
-        UnsupportedOperationException."""
-        self._data: TPMap[str, Any] = pyrsistent.m()
-        self._snapshot = copy.copy(snapshot) if snapshot else None
+    def __init__(self, snapshot: Optional["Snapshot"] = None) -> None:
+        self._realization_states = defaultdict(dict)
+        """A shallow dictionary of realization states. The key is a string with
+        realization number, pointing to a dict with members active (bool),
+        start_time (datetime), end_time (datetime) and status (str)."""
 
-    def update_status(self, status: str) -> None:
-        self._apply_update(SnapshotDict(status=status))
+        self._step_states = defaultdict(dict)
+        """A shallow dictionary of step states. The key is a tuple of two strings with
+        realization id and step id, pointing to a dict with the same members as the Step
+        class, except Jobs"""
 
-    def update_metadata(self, metadata: Dict[str, Any]) -> None:
-        if self._snapshot is None:
-            raise UnsupportedOperationException(
-                f"updating metadata on {self.__class__} without providing a snapshot"
-                + " is not supported"
-            )
-        dictionary = pyrsistent.pmap({ids.METADATA: metadata})
-        self._data = _recursive_update(self._data, dictionary, check_key=False)
-        self._snapshot.merge_metadata(metadata)
+        self._job_states = defaultdict(dict)
+        """A shallow dictionary of job states. The key is a tuple of three
+        strings with realization id, step id and job id, pointing to a dict with
+        the same members as the Job."""
 
-    def update_real(
-        self,
-        real_id: str,
-        real: "RealizationSnapshot",
-    ) -> None:
-        self._apply_update(SnapshotDict(reals={real_id: real}))
+        self._ensemble_state: Optional[str] = None
+        self._metadata = defaultdict(dict)
 
-    def _apply_update(self, update: "SnapshotDict") -> None:
-        if self._snapshot is None:
-            raise UnsupportedOperationException(
-                f"trying to mutate {self.__class__} without providing a snapshot is "
-                + "not supported"
-            )
-        dictionary = update.dict(
-            exclude_unset=True, exclude_none=True, exclude_defaults=True
+        self._snapshot = snapshot
+
+    @property
+    def status(self) -> str:
```

Comment:
This property seems to not be used.

Reply:
it's used in `_builder/_ensemble.py`

Reply:
something is funky with the coverage calc - [here](https://app.codecov.io/gh/equinor/ert/pull/5828/blob/src/ert/ensemble_evaluator/_builder/_ensemble.py#L148), one usage of the method seems covered:

Reply:
something funky, maybe mocking only. we should stop mocking.